### PR TITLE
feat(api): support air gap command in V3+V4 executors

### PR DESCRIPTION
**Pull request by IanLondon**
# Overview

We finally will implement air gap! This PR should make the executors for both V3 and V4 protocols support the `"airGap"` command in the schema (command is the same in both schemas)

# Changelog
- Support `airGap` JSON command

# Review requests

- [ ] Careful code review plz
- [ ] Test on robot: air gap to the current well
- [ ] Test on robot: air gap to a new well

This is tricky to test correctly, because the API's `InstrumentContext.air_gap` fn is stateful. However, the JSON command should be stateless.

Like asp/disp/etc, API `air_gap` uses a remembered flow rate that must be reset (but this we already solved for asp/disp/etc). New problem for air gap is that `air_gap` uses the previous location and does not have an argument that allows specifying a well/location. However, the JSON command has a `well` parameter. Additional surface for error is that `air_gap` takes a height from top, while the JSON command specifies an `offsetFromBottomMm`, so I might have messed up that geometric transform. This should be captured in the tests, but I'm not familiar enough with the `Location` class stuff to think I got it right my first try!

TL;DR -- when testing on robot, make sure the air gap doesn't do a little dance up/down in Z (due to the `move_to`), and make sure the air gap's `offsetFromBottomMm` translates to that specified position from the well bottom.

Here is a test V4 protocol to run on the robot: [airGapTest.json.zip](https://github.com/Opentrons/opentrons/files/4832856/airGapTest.json.zip)

It should:
- Pick up tip from B1
- Aspirate 5uL from source plate in slot 2, well A1. (This is at 2mm from bottom.)
- Air gap at different well of source plate, well A2. The air gap should happen 30mm from the well bottom, which should be 5 mm above the well top (well depth is 25). 
- Dispense 4.5uL in well B1 of the other destination plate
- Drop the tip

If you need help editing the protocol to match a labware that you have (this is an 8-well plate, probably uncommon), LMK and I can rewrite the protocol for you.

# Risk assessment

This shouldn't affect any existing features, but if we get it wrong it's a pain to bugfix and re-release later.

See above for risky business

### CODE REVIEW

File: api/src/opentrons/protocol_api/execute_v3.py [hunk 1]
```diff
@@ -153,6 +153,25 @@ def _dispense(instruments: Dict[str, InstrumentContext],
...
+    volume = params['volume']
+    _set_flow_rate(pipette, params)
+    well = _get_well(loaded_labware, params)
+    offset_from_top = offset_from_bottom - well._depth
+
+    # NOTE(IL, 2020-06-25): air_gap API fn is stateful, uses location
+    # cache. The JSON atomic command should be stateless. We'll
+    # explicitly move_to the specified well to ensure the location
+    # cache is set to that well.
+    pipette.move_to(well.top().move(Point(0, 0, offset_from_top)))
```
Comment (Laura-Danielle) [inline api/src/opentrons/protocol_api/execute_v3.py:171]: Since you don't ever need the result of `well.top()`, you could just pass in the `offset_from_top` value directly into `well.top()` here.

File: api/src/opentrons/protocol_api/execute_v3.py [hunk 2]
```diff
@@ -153,6 +153,25 @@ def _dispense(instruments: Dict[str, InstrumentContext],
     pipette.dispense(volume, location)
 
 
+def _air_gap(instruments: Dict[str, InstrumentContext],
+             loaded_labware: Dict[str, labware.Labware],
+             params: 'StandardLiquidHandlingParams') -> None:
```
Comment (sfoster1) [inline api/src/opentrons/protocol_api/execute_v3.py:158]: [nitpick] don't need an explicit `None` return type annotation

Comment (IanLondon) [reply api/src/opentrons/protocol_api/execute_v3.py:158]: as discussed, will do that to the while file in a separate PR
